src/scanner/images/skopeo.ts

```diff
@@ -57,30 +78,38 @@ export async function pull(
     sanitise: false,
   });
 
-  await pullWithRetry(args, destination, workloadName);
+  const env: Record<string, string | undefined> = {
+    // The Azure CR credentials helper requires these env vars:
+    AZURE_CLIENT_ID: process.env.AZURE_CLIENT_ID,
+    AZURE_TENANT_ID: process.env.AZURE_TENANT_ID,
+    AZURE_FEDERATED_TOKEN_FILE: process.env.AZURE_FEDERATED_TOKEN_FILE,
+    AZURE_FEDERATED_TOKEN: process.env.AZURE_FEDERATED_TOKEN,
+    AZURE_AUTHORITY_HOST: process.env.AZURE_AUTHORITY_HOST,
+  };
+  await pullWithRetry(args, env, destination, workloadName);
+
+  return await extractImageDigests(
+    prefixRespository(image, SkopeoRepositoryType.ImageRegistry),
+    env,
+  );
 }
 
 async function pullWithRetry(
   args: Array<processWrapper.IProcessArgument>,
+  env: Record<string, string | undefined>,
   destination: string,
   workloadName: string,
 ): Promise<void> {
   const MAX_ATTEMPTS = 10;
   const RETRY_INTERVAL_SEC = 0.2;
 
-  for (let attempt = 1; attempt <= MAX_ATTEMPTS; attempt++) {
+  let attempt = 1;
```

Comment:
This refactor from `for` to `while` was to support returning something other than `void` here, but now we're not doing that, it might be worth putting it back how it was to avoid the chance of any accidental regressions here.
Passing `env` in still makes sense because we're sharing it with the other call now.

I do _like_ that we're not returning the digests from this function anymore, nice one!